ladybug/epw.py

```diff
@@ -241,57 +219,49 @@ def _get_data_by_field(self, field_number):
+        """
         # load data if it's  not loaded
         if not self.is_data_loaded:
             self._import_data()
 
         # write the file
-        with open(full_path, writemode) as modEpwFile:
-            modEpwFile.writelines(self._header)
-            lines = []
-            try:
-                # move first item to end position for fields on the hour
-                for field in range(0, self._num_of_fields):
-                    middle_hour = EPWFields.field_by_number(field).middle_hour
-                    if middle_hour is False:
-                        first_hour = self._data[field].pop(0)
-                        self._data[field].append(first_hour)
-
-                for hour in xrange(0, 8760):
-                    line = []
-                    for field in range(0, self._num_of_fields):
-                        line.append(str(self._data[field].data[hour].value))
-                    lines.append(",".join(line) + "\n")
-            except IndexError:
-                # cleaning up
-                modEpwFile.close()
-                length_error_msg = 'Data length is not 8760 hours and cannot be ' + \
-                    'saved as an EPW file.'
-                raise ValueError(length_error_msg)
-            else:
-                modEpwFile.writelines(lines)
-            finally:
-                del(lines)
-                # move last item to start position for fields on the hour
+        lines = self._header
+        try:
+            # move first item to end position for fields on the hour
+            for field in range(0, self._num_of_fields):
+                middle_hour = EPWFields.field_by_number(field).middle_hour
+                if middle_hour is False:
+                    first_hour = self._data[field].pop(0)
+                    self._data[field].append(first_hour)
+
+            for hour in xrange(0, 8760):
+                line = []
                 for field in range(0, self._num_of_fields):
-                    middle_hour = EPWFields.field_by_number(field).middle_hour
-                    if middle_hour is False:
-                        last_hour = self._data[field].pop()
-                        self._data[field].insert(0, last_hour)
+                    line.append(str(self._data[field].data[hour].value))
+                lines.append(",".join(line) + "\n")
+        except IndexError:
+            # cleaning up
+            length_error_msg = 'Data length is not 8760 hours and cannot be ' + \
+                'saved as an EPW file.'
+            raise ValueError(length_error_msg)
+        else:
+            file_data = ''.join(lines)
+            write_to_file(file_path, file_data, True)
+        finally:
+            del(lines)
+            # move last item to start position for fields on the hour
+            for field in range(0, self._num_of_fields):
+                middle_hour = EPWFields.field_by_number(field).middle_hour
```

Comment:
Can you remind me what does `middle_hour` return? Is it the one that says if the data is shifted for an hour? why did we name it middle_hour?

Reply:
The middle_hour property tells you which of the EPW data types are recorded on the hour (when middle_hour is Flase) and which ones are represent a cumulative value of the previous hour (when middle_hour is True).  In this case it's being used to decide whether you need to take the last item of the EPW and put it in the spot to January 1 at 00:00.

Reply:
Makes sense. I would still vote for renaming it to a more explicit name.